stock_getter/backend_stockgetter/stock_service/forecast_tools/forecaster.py

```diff
@@ -0,0 +1,149 @@
+###################################################################
+# This file defines the Forecaster object, used to wrap forecasting
+# models
+###################################################################
+from typing import Any, Dict, List, Optional, Tuple, Union
+import pandas as pd
+from .forecast_models import ProphetModel
+
+
+###################################################################
+# Forecaster Class
+###################################################################
+class Forecaster():
+    """Forecaster wrapper class to manage forecasting via different models.
+    This does data preprocessing and than uses an object for a given model
+    type (for now just prophet) to produce forecasts
+    """
+    def __init__(self) -> None:
+        """For now do nothing in constructor"""
+
+    def process_stock_data(self, stock_data: dict, grain:str) -> dict:
+        """Method to pre-process data into Records orientation needed
+        for the front end
+
+        Args:
+            stock_data (dict): stock data in dict format
+            grain (str): data grain ('weekly', 'daily', 'monthly')
+
+        Returns:
+            dict: dict with stock data in records orientation
+        """
+        data_df = self.parse_stock_data(stock_data, grain)
+        ret_dict = self.convert_data_to_records_orientation(data_df)
+        return ret_dict
+
+    def forecast(self, stock_data: dict, grain: str, horizon: str,
+                 model_type: str="Prophet") -> dict:
+        """Method to generate forecast
+
+        Args:
+            stock_data (dict): stock data in dict format
+            grain (str): data grain ('weekly', 'daily', 'monthly')
+            horizon (str): forecast horizon
+            model_type(str): forecasting model type. For now just 
+                supports Facebook prophet algorithm:
+                https://facebook.github.io/prophet/
+
+        Returns:
+            dict: dict with the data (in records orientation)
+        """
+        data_df = self.parse_stock_data(stock_data, grain)
+        
+        # Instantiate appropriate model (for now just prophet)
+        if model_type == "Prophet":
+            forecast_model = ProphetModel(data_df, grain)
+        else:
+            msg = "***ERROR: incorrect model type specified"
```

Comment:
Think we can have a single string for this. Also right now it is missing an empty space between the two strings.

Reply:
I am not sure what you mean. Can you please clarify?

Reply:
This line is creating a string msg and the one below is concatenating to it right away. We can just have:
msg = "***ERROR: incorrect model type specified returning just history with no forecast"

Also concatenating strings currently has no space between the word "specified" and "returning"

Reply:
This is done to stay within the 79 line length per PEP8. This is standard practice to create long strings and keep them within the 79 character line length.